Multisig/UI/Transaction/ExecuteTransaction/TransactionExecutionController.swift

```diff
@@ -234,6 +234,16 @@ class TransactionExecutionController {
                     //
                     // to avoid that, we handle both cases in one do-catch block.
                     gas = try partialResults.gas.get()
+                    
+                    // GH-3084: fix for a bug in Nethermind requiring 30% increase in the gas estimation
+                    // on gnosis chain for transactions with positive safeTxGas
+                    if chain.id == Chain.ChainID.gnosis,
+                       let safeTxGas = transaction.multisigInfo?.safeTxGas.value,
+                       safeTxGas > 0 {
+                        // `gas` is UInt64, so it can overflow if a large number returned from the estimator
+                        let (newGas, overflow) = gas!.multipliedReportingOverflow(by: 130)
+                        gas = (overflow ? .max : newGas) / 100
```

Comment:
I know it's an edge case but I think this is wrong, imagine if gas > .max/100, this way we have reduced the gas instead of increase

Reply:
the other way is be to temp convert to u256, make an operation, and then limit with higher bound at u64 max;

    gas = u64( u256(gas) * 130 / 100 )

    would handle all cases

yet another way to add a 1/3 of the value;

    let (newGas, overflow) = gas + gas / 3
    gas = overflow ? .max : newGas

    ^^ for gas < 3 will result in no change;